imap/spool.c

```diff
@@ -207,6 +207,30 @@ static int parseheader(struct protstream *fin, FILE *fout,
             break;
 
         case BODY_START:
+            /* leading whitespace might be folded too */
+            if (c == '\r' || c == '\n') {
+                int peek;
```

Comment:
This dealing-with-newlines section is more-or-less a direct copy of the same code in the `BODY:` state immediately below. I've deliberately left the style the same so that it's clear to the casual reader that they're the same.